### PR TITLE
fix: dedup groupby in viz.py while preserving order

### LINKED ISSUE #10218: Preserve Column Order in Table CSV Export
I am experiencing this new issue with superset 36. 

I am pretty sure in the previous versions of superset Table exported in CSV had column order preserved. 
I also have trouble exporting CSV data with 'latin-1' encoding, which I did not have before 

Is this possible or I am dreaming and it never worked ? :/

### PR DESCRIPTION
**Pull request by mistercrunch**
closes https://github.com/apache/incubator-superset/issues/10218

### CODE REVIEW

File: superset/tasks/slack_util.py [hunk 1]
```diff
@@ -37,6 +35,7 @@ def deliver_slack_msg(
     body: str,
     file: Optional[Union[str, IOBase, bytes]],
 ) -> None:
+    config = current_app.config
```
Comment (mistercrunch) [inline superset/tasks/slack_util.py:38]: bycatch, the mypy commit hook was triggering on this...

File: superset/viz.py [hunk 2]
```diff
@@ -322,7 +322,8 @@ def query_obj(self) -> QueryObjectDict:
         gb = self.groupby
         metrics = self.all_metrics or []
         columns = form_data.get("columns") or []
-        groupby = list(set(gb + columns))
+        # merge list and dedup while preserving order
+        groupby = list(OrderedDict.fromkeys(gb + columns))
```
Comment (villebro) [inline superset/viz.py:326]: Nit, as of 3.7 key order is now officially guaranteed (already implicitly so in 3.6), so no need to use `OrderedDict` any more: 
```suggestion
        groupby = list(dict.fromkeys(gb + columns))
```
Reference: https://docs.python.org/3.7/tutorial/datastructures.html#dictionaries
> Performing `list(d)` on a dictionary returns a list of all the keys used in the dictionary, in insertion order

**Review by villebro — APPROVED** [review]
LGTM, non-blocking nit

**Comment by villebro** [comment]
Merging this for upcoming `0.37.1`

**Pull request merged**